src/Evaluator.ml

```diff
@@ -68,6 +74,7 @@ let rec subst_m term identm Location.{ data = body; _ } =
                   boxed = subst_m term identm boxed;
                   body = subst_m term identm body;
                 } )
+  | Match _ -> Location.locate body
```

Comment:
Here too, we need to walk the expression we match on and the branches.

Reply:
Done